core/extension/gdextension_interface.json

```diff
@@ -0,0 +1,9260 @@
+            "return_value": {
+                "type": "void"
+            },
+            "arguments": [
+                {
+                    "name": "p_instance",
+                    "type": "GDExtensionClassInstancePtr"
+                },
+                {
+                    "name": "p_name",
+                    "type": "GDExtensionConstStringNamePtr"
+                },
+                {
+                    "name": "p_virtual_call_userdata",
+                    "type": "void*"
+                },
+                {
+                    "name": "p_args",
+                    "type": "const GDExtensionConstTypePtr*"
+                },
+                {
+                    "name": "r_ret",
+                    "type": "GDExtensionTypePtr"
+                }
+            ]
+        },
+        {
+            "name": "GDExtensionClassCreationInfo",
+            "kind": "struct",
+            "members": [
+                {
+                    "name": "is_virtual",
+                    "type": "GDExtensionBool"
+                },
+                {
+                    "name": "is_abstract",
+                    "type": "GDExtensionBool"
+                },
+                {
+                    "name": "set_func",
+                    "type": "GDExtensionClassSet"
+                },
+                {
+                    "name": "get_func",
+                    "type": "GDExtensionClassGet"
+                },
+                {
+                    "name": "get_property_list_func",
+                    "type": "GDExtensionClassGetPropertyList"
+                },
+                {
+                    "name": "free_property_list_func",
+                    "type": "GDExtensionClassFreePropertyList"
+                },
+                {
+                    "name": "property_can_revert_func",
+                    "type": "GDExtensionClassPropertyCanRevert"
+                },
+                {
+                    "name": "property_get_revert_func",
+                    "type": "GDExtensionClassPropertyGetRevert"
+                },
+                {
+                    "name": "notification_func",
+                    "type": "GDExtensionClassNotification"
+                },
+                {
+                    "name": "to_string_func",
+                    "type": "GDExtensionClassToString"
+                },
+                {
+                    "name": "reference_func",
+                    "type": "GDExtensionClassReference"
+                },
+                {
+                    "name": "unreference_func",
+                    "type": "GDExtensionClassUnreference"
+                },
+                {
+                    "name": "create_instance_func",
+                    "type": "GDExtensionClassCreateInstance",
+                    "description": [
+                        "(Default) constructor; mandatory. If the class is not instantiable, consider making it virtual or abstract."
```

Comment:
Since this field is mandatory, would it be useful to add something like a "mandatory" bool so that extensions can handle it differently?  For example, when initializing the info struct, this field could exclude a default value to make it obvious to the extension user that this function needs to be provided.